journal/src/test/java/io/zeebe/journal/file/JournalTest.java

```diff
@@ -497,6 +508,77 @@ public void shouldInvalidateAllEntries() throws Exception {
     assertThat(reader.hasNext()).isFalse();
   }
 
+  @Test
+  public void shouldIgnorePartiallyWrittenRecordOnStartup() throws Exception {
+    // given
+    data.wrap("000".getBytes(StandardCharsets.UTF_8));
+    final var firstRecord = journal.append(data);
+    final var secondRecord = journal.append(data);
+    assertThat(directory.toFile().listFiles()).hasSize(1);
+    assertThat(directory.toFile().listFiles()[0].listFiles()).hasSize(1);
+    final File log = directory.toFile().listFiles()[0].listFiles()[0];
+
+    final File tempFile = new File(temp.toFile(), "rand");
+    assertThat(tempFile.createNewFile()).isTrue();
+    assertThat(corruptFile(tempFile, log, secondRecord.index())).isTrue();
+    journal.close();
+
+    // when/then
+    journal = openJournal();
```

Comment:
This test can be split into two 
1. after restart lastIndex == firstRecord.index()
2. After restart it can still write new records
In both cases use readers to verify the states, as there could be bugs in the writer which gets unnoticed if only rely on the return value of the append.

Reply:
Why would we separate the two assertions? The assertions check two parts of the state to verify that the same behavior was taken after the same situation

Reply:
It would be nice to have small tests that tests one condition. To me it makes sense to have two tests - `shouldTruncateCorruptedEntries`, `shouldWriteNewEntriesAfterCorruptedEntriesDetected`. 
Anyway, whether we split the test or not, the assertions should be done using readers.